File: sdks/java/src/main/java/com/puxbay/resources/ReturnsResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.ReturnRequest;

public class ReturnsResource {
    private final Puxbay client;

    public ReturnsResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<ReturnRequest> list(int page) throws PuxbayException {
        return (PaginatedResponse<ReturnRequest>) client.request("GET", "returns/?page=" + page, null, PaginatedResponse.class);
    }

    public ReturnRequest get(String returnId) throws PuxbayException {
        return client.request("GET", "returns/" + returnId + "/", null, ReturnRequest.class);
    }

    public ReturnRequest create(ReturnRequest returnData) throws PuxbayException {
        return client.request("POST", "returns/", returnData, ReturnRequest.class);
    }
    
    public ReturnRequest approve(String returnId) throws PuxbayException {
        return client.request("POST", "returns/" + returnId + "/approve/", null, ReturnRequest.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/ReturnsResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.ReturnRequest
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class ReturnsResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1): PaginatedResponse<ReturnRequest> =
        client.request(HttpMethod.Get, "returns/?page=$page")
    
    suspend fun get(returnId: String): ReturnRequest =
        client.request(HttpMethod.Get, "returns/$returnId/")
    
    suspend fun create(returnData: ReturnRequest): ReturnRequest =
        client.request(HttpMethod.Post, "returns/", returnData)
    
    suspend fun approve(returnId: String): ReturnRequest =
        client.request(HttpMethod.Post, "returns/$returnId/approve/")
}
